core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1795,10 +1803,10 @@ class UnifiedLog(@volatile var logStartOffset: Long,
    * remove deleted log metrics
    */
   private[log] def removeLogMetrics(): Unit = {
-    removeMetric(LogMetricNames.NumLogSegments, tags)
-    removeMetric(LogMetricNames.LogStartOffset, tags)
-    removeMetric(LogMetricNames.LogEndOffset, tags)
-    removeMetric(LogMetricNames.Size, tags)
+    metricNames.foreach {
+      case (name, tags) => removeMetric(name, tags)
+    }
+    metricNames = Map.empty
```

Comment:
Could you explain why we need this change to remove metrics, instead of the original one?

Reply:
Is it because when we tried to removeMetrics, the tags might become "non-future"?

Reply:
> Is it because when we tried to removeMetrics, the tags might become "non-future"?

yep. The tags DOES NOT include "future" after the log dir get renamed. Hence, we have to keep previous metrics name in order to delete correct metrics.